ui/selectors/selectors.js

```diff
@@ -706,17 +705,117 @@ export function getTargetSubjectMetadata(state, origin) {
   return metadata;
 }
 
+///: BEGIN:ONLY_INCLUDE_IN(snaps)
+/**
+ * Input selector for reusing the same state object.
+ * Used in memoized selectors created with createSelector
+ * when raw state is needed to be passed to other selectors
+ * used to achieve re-usability.
+ *
+ * @param state - Redux state object.
+ * @returns Object - Redux state object.
+ */
+export const rawStateSelector = (state) => state;
+
+/**
+ * Input selector used to retrieve Snaps that are added to Snaps Directory.
+ *
+ * @param state - Redux state object.
+ * @returns Object - Containing verified Snaps from the Directory.
+ */
+const selectVerifiedSnapsRegistry = (state) =>
+  state.metamask.database.verifiedSnaps;
+
+/**
+ * Input selector providing a way to pass a snapId as an argument.
+ *
+ * @param _state - Redux state object.
+ * @param snapId - ID of a Snap.
+ * @returns string - ID of a Snap that can be used as input selector.
+ */
+const selectSnapId = (_state, snapId) => snapId;
+
+/**
+ * Input selector for retrieving all installed Snaps.
+ *
+ * @param state - Redux state object.
+ * @returns Object - Installed Snaps.
+ */
+export const selectInstalledSnaps = (state) => state.metamask.snaps;
+
 /**
  * Retrieve registry data for requested Snap.
  *
  * @param state - Redux state object.
  * @param snapId - ID of a Snap.
  * @returns Object containing metadata stored in Snaps registry for requested Snap.
  */
-export function getSnapRegistryData(state, snapId) {
-  const snapsRegistryData = state.metamask.database.verifiedSnaps;
-  return snapsRegistryData ? snapsRegistryData[snapId] : null;
-}
+export const getSnapRegistryData = createSelector(
+  [selectVerifiedSnapsRegistry, selectSnapId],
+  (snapsRegistryData, snapId) => {
+    return snapsRegistryData ? snapsRegistryData[snapId] : null;
+  },
+);
+
+/**
+ * Find and return Snap's latest version available in registry.
+ *
+ * @param state - Redux state object.
+ * @param snapId - ID of a Snap.
+ * @returns String SemVer version.
+ */
+export const getSnapLatestVersion = createSelector(
+  [getSnapRegistryData],
+  (snapRegistryData) => {
+    if (!snapRegistryData) {
+      return null;
+    }
+
+    return Object.keys(snapRegistryData.versions).reduce((latest, version) => {
+      return semver.gt(version, latest) ? version : latest;
+    }, '0.0.0');
+  },
+);
+
+/**
+ * Return a Map of all installed Snaps with available update status.
+ *
+ * @param state - Redux state object.
+ * @returns Map Snap IDs mapped to a boolean value (true if update is available, false otherwise).
+ */
+export const getAllSnapsAvailableUpdateStatus = createSelector(
```

Comment:
```suggestion
export const getAllSnapAvailableUpdates = createSelector(
```
I think this is slightly more clear?

Reply:
Updated / refactored.